File: largest-3-same-digit-number-in-string.ts
{
    var largestGoodInteger = function(num) {
        let max = ""

        for(let i = 2; i < num.length; i++) {
            if(num[i] === num[i - 1] && num[i] === num[i - 2]) {
                const sub = num[i].repeat(3)
                if(sub > max) {
                    max = sub
                }
            }
        }

        return max
    };

    console.log(largestGoodInteger("99921234324444"))
}

File: largest-3-same-digit-number-in-string.rs
pub fn largest_good_integer(num: String) -> String {
    match num
        .as_bytes()
        .chunk_by(u8::eq)
        .filter(|xs| xs.len() >= 3)
        .max()
    {
        Some(xs) => [xs[0] as char; 3].iter().collect(),
        _ => "".into(),
    }
}
